src/topology/continuous_function/compact.lean

```diff
@@ -427,4 +427,71 @@ end
 
 end weierstrass
 
+
+/-!
+### Star structures
+
+In this section, if `β` is a normed ⋆-group, then so is the space of
+continuous functions from `α` to `β`, by using the star operation pointwise.
+
+If `𝕜` is normed field and a ⋆-ring over which `β` is a normed algebra and a
+star module, then the space of continuous functions from `α` to `β`
+is a star module.
+
+If `β` is a ⋆-ring in addition to being a normed ⋆-group, then `C(α, β)`
+inherits a ⋆-ring structure.
+
+Furthermore, if `α` is compact and `β` is a C⋆-ring, then `C(α, β)` is a C⋆-ring.  -/
+
+section normed_space
+
+variables {α : Type*} {β : Type*} {𝕜 : Type*}
+variables [normed_field 𝕜] [star_ring 𝕜]
+variables [topological_space α] [normed_group β] [star_add_monoid β]
+  [normed_star_group β] [normed_space 𝕜 β] [star_module 𝕜 β]
+
+lemma _root_.bounded_continuous_function.mk_of_compact_star [compact_space α] (f : C(α, β)) :
+  mk_of_compact (star f) = star (mk_of_compact f) := rfl
+
+instance [compact_space α] : normed_star_group C(α, β) :=
+{ norm_star := λ f, by rw [←bounded_continuous_function.norm_mk_of_compact,
+                          bounded_continuous_function.mk_of_compact_star, norm_star,
+                          bounded_continuous_function.norm_mk_of_compact] }
+
+instance : star_module 𝕜 C(α, β) :=
+{ star_smul := λ k f, by { ext, exact star_smul _ _ } }
+
+end normed_space
+
+section cstar_ring
+
+variables {α : Type*} {β : Type*} {𝕜 : Type*}
+variables [normed_field 𝕜] [star_ring 𝕜]
+variables [topological_space α] [normed_ring β] [star_ring β]
+
+instance [has_continuous_star β] : star_ring C(α, β) :=
```

Comment:
This one can probably move to the other file too

Reply:
Thanks!